### PR TITLE
WRP-7826: Upgraded eslint-plugin-react to v7.32.2 and created new eslint.config.js

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop [stanca.pop@lgepartner.com](mailto:stanca.pop@lgepartner.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
_eslint-plugin-react_ v7.32.0 has been released on Jan 11, 2023. This version has several breaking changes, including jsx enable config being moved under languageOptions.
This new schema change cannot work on the .eslintrc.js file, only works on the eslint.config.js file.  
`.eslintrc.js` will be deprecated from eslint v9, only `eslint.config.js` file will be supported.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Upgraded _eslint-plugin-react_ and replaced `.eslintrc.js` with `eslint.config.js`. 

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRP-7826

### CODE REVIEW

File: commands/eject.js [hunk 1]
```diff
@@ -216,7 +216,7 @@ function configurePackage(bare) {
 	app.eslintConfig = eslintConfig;
 	app.eslintIgnore = app.eslintIgnore || [];
 	app.eslintIgnore = app.eslintIgnore.concat(eslintIgnore.filter(l => !app.eslintIgnore.includes(l)));
-	backupOld(['.eslintignore', '.eslintrc.js', '.eslintrc.yaml', '.eslintrc.yml', '.eslintrc.json', '.eslintrc']);
+	backupOld(['.eslintignore', 'eslint.config.js', '.eslintrc.yaml', '.eslintrc.yml', '.eslintrc.json', '.eslintrc']);
```
Comment (daniel-stoian-lgp) [inline commands/eject.js:219]: i'm not sure what to do about the other file formats. I'll need someone else's opinion

File: eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const enactConfig = require('eslint-config-enact');
+const importPlugin = require('eslint-plugin-import');
+const prettierPlugin = require('eslint-plugin-prettier');
+const prettierRecommended = require('eslint-config-prettier');
```
Comment (daniel-stoian-lgp) [inline eslint.config.js:4]: you mentioned "but added prettier/recommended rules manually and it seems to work (source: https://github.com/prettier/eslint-plugin-prettier#recommended-configuration)."
Is this duplicate import still needed?
If not, let's just define `"const prettier = require('eslint-plugin-prettier');"`

**Review by daniel-stoian-lgp — COMMENTED** [review]
Please check my comments

File: commands/eject.js (hunk 1 above)
Comment (stanca-pop-lgp) [reply commands/eject.js:219]: Yes, I would appreciate someone else's opinion as well.

**Review by daniel-stoian-lgp — APPROVED** [review]

File: commands/eject.js (hunk 1 above)
Comment (hong6316) [reply commands/eject.js:219]: backupOld() is the function that if parameters(file names) exist in the project, saves the backup file by attaching *.old.
I recommend keeping .eslintrc.js and adding eslint.config.js to the parameter.

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+					'newlines-between': 'never',
+					groups: ['builtin', 'external', 'internal', 'parent', 'sibling', 'index']
+				}
+			],
+			'prettierPlugin/prettier': 'error',
+			'prettierConfig/arrow-body-style': 'off',
+			'prettierConfig/prefer-arrow-callback': 'off'
+		}
+	}
+];
```
Comment (hong6316) [inline eslint.config.js:42]: Could you explain the changed config file in more detail in "Resolution"?
I compared these configurations with the deleted .eslintrc.js, and there seem to be many changes.

**Comment by stanca-pop-lgp** [comment]
Closing in favor of https://github.com/enactjs/cli/pull/312.

**Pull request closed without merging**